atomic_reactor/plugins/pre_pin_operator_digest.py

```diff
@@ -229,10 +249,70 @@ def replace_registry(self, image):
             return image
         return self._replace(image, registry=self.registry_replace[image.registry])
 
+    # TODO: rename, because it can also replace namespace
+    def replace_repo(self, image):
+        """
+        Replace image repo based on OSBS site/user configuration for image registry
+
+        :param image: ImageName
+        :return: ImageName
+        """
+        package_mapping = self._get_package_mapping(image.registry)
+        if package_mapping is None:  # repo_replacements not configured for registry
+            return image
+
+        replacement = package_mapping.get(image.repo)
+
+        if replacement is None:
+            raise RuntimeError(
+                "repo_replacements configured for registry but repo is missing: {}".format(image)
+            )
+        elif isinstance(replacement, list):
+            replacements = ", ".join(replacement)
+            raise RuntimeError(
+                "Multiple options found when replacing repo in {}: {}".format(image, replacements)
+            )
+
+        # TODO: replacement can (probably should) be namespace/repo
```

Comment:
Replacement is package name --> namespace/repo. We should strip everything from original url.
Package name must be taken from LABELs in image (com.redhat.bzcomponent or so), so registry has to be queried for that.